pkg/client/record/event.go

```diff
@@ -111,19 +114,26 @@ func (eventBroadcaster *eventBroadcasterImpl) StartRecordingToSink(sink EventSin
 			// Events are safe to copy like this.
 			eventCopy := *event
 			event = &eventCopy
-
+			var patch []byte
 			previousEvent := eventCache.getEvent(event)
 			updateExistingEvent := previousEvent.Count > 0
 			if updateExistingEvent {
-				event.Count = previousEvent.Count + 1
-				event.FirstTimestamp = previousEvent.FirstTimestamp
+				// we still need to copy Name because the Patch relies on the Name to find the target event
 				event.Name = previousEvent.Name
-				event.ResourceVersion = previousEvent.ResourceVersion
+				event.Count = previousEvent.Count + 1
+
+				// we need to make sure the Count and LastTimestamp are the only differences between event and the eventCopy2
```

Comment:
I was hoping using an existing method would reduce the amount of code here, but it increased it :) Oh well